File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/search/SearchPresenterImpl.java
package com.twtstudio.tjwhm.lostfound.search;

import com.twt.wepeiyang.commons.network.RetrofitProvider;
import com.twt.wepeiyang.commons.network.RxErrorHandler;
import com.twtstudio.tjwhm.lostfound.waterfall.WaterfallBean;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by tjwhm on 2017/7/6.
 **/

public class SearchPresenterImpl implements SearchContract.SearchPresenter {
    SearchContract.SearchUIView searchUIView;
    SearchApi searchApi;

    public SearchPresenterImpl(SearchContract.SearchUIView searchUIView) {
        this.searchUIView = searchUIView;
    }

    @Override
    public void loadSearchData(String keyword, int page) {
        searchApi = RetrofitProvider.getRetrofit().create(SearchApi.class);
        searchApi.loadSearchData(keyword, String.valueOf(page))
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(this::setSearchData, new RxErrorHandler());
    }

    @Override
    public void setSearchData(WaterfallBean waterfallBean) {
        searchUIView.setSearchData(waterfallBean);
    }
}

File: WePeiYang/lostfound2/src/main/java/com/example/lostfond2/search/SearchPresenterImpl.kt
package com.example.lostfond2.search

import com.example.lostfond2.service.LostFoundService
import com.example.lostfond2.service.MyListDataOrSearchBean
import com.twt.wepeiyang.commons.experimental.extensions.QuietCoroutineExceptionHandler
import com.twt.wepeiyang.commons.experimental.extensions.awaitAndHandle
import com.twt.wepeiyang.commons.experimental.network.CommonBody
import kotlinx.coroutines.experimental.android.UI
import kotlinx.coroutines.experimental.async
import kotlinx.coroutines.experimental.launch

class SearchPresenterImpl(val searchUIView: SearchContract.SearchUIView) : SearchContract.SearchPresenter {
    override fun loadSearchData(keyword: String, page: Int) {
        launch(UI + QuietCoroutineExceptionHandler) {
            val searchList: CommonBody<List<MyListDataOrSearchBean>> = LostFoundService.getSearch(keyword, page).await()

            if (searchList.error_code == -1) {
                setSearchData(searchList.data!!)
            }
        }

    }

    override fun setSearchData(waterfallBean: List<MyListDataOrSearchBean>) {
        searchUIView.setSearchData(waterfallBean)
    }
}